node/network/pov-distribution/src/lib.rs

```diff
@@ -75,23 +89,45 @@ impl<C> Subsystem<C> for PoVDistribution
 	}
 }
 
+#[derive(Default)]
 struct State {
+	/// A state of things going on on a per-relay-parent basis.
 	relay_parent_state: HashMap<Hash, BlockBasedState>,
+
+	/// Info on peers.
 	peer_state: HashMap<PeerId, PeerState>,
+
+	/// Our own view.
 	our_view: View,
+
+	/// Connect to relevant groups of validators at different relay parents.
+	connection_requests: validator_discovery::ConnectionRequests,
+
+	/// Metrics.
 	metrics: Metrics,
 }
 
 struct BlockBasedState {
 	known: HashMap<Hash, Arc<PoV>>,
+
 	/// All the PoVs we are or were fetching, coupled with channels expecting the data.
 	///
 	/// This may be an empty list, which indicates that we were once awaiting this PoV but have
 	/// received it already.
-	fetching: HashMap<Hash, Vec<oneshot::Sender<Arc<PoV>>>>,
+	fetching: HashMap<Hash, PerFetchedPoV>,
+
 	n_validators: usize,
 }
 
+/// Information kept about each PoV we attempt to fetch.
+struct PerFetchedPoV {
+	/// Channels expecting the data.
+	///
+	/// This may be an empty list, which indicates that we were once awaiting this PoV
+	/// but have received it already.
```

Comment:
I think this has some duplication with the `fetching` field above. Do you think we should deduplicate?

Reply:
Yep already did that